### PR TITLE
Refactor and test `nitrification_volatilization.py`

### LINKED ISSUE #484: refactor/test nitrification_volatilization.py (soil/nitrogen_cycling) into SC_redesign directory

### PR DESCRIPTION
**Pull request by Isgotkowitz**
# Summary
This PR refactors and tests the `nitrification_volatilization.py` file, which handles the nitrification and volatilization of ammonium within the soil profile. The `NitrificationVolatilization` file closely follows the SWAT section 3:1.3, Nitrification and Ammonia Volatilization.

# Main Changes  
- `nitrification_volatilization.py`:
  - This file consists of one main method (`do_daily_nitrification_and_volatilization()`) to be called by higher-level modules, and 8 `@staticmethod`s
  - All methods in this module are tested in `test_nitrification_volatilization.py`
- `layer_data.py`:
  - Two attributes have been added to use in `NitrificationVolatilization` 
    1) `ammonium_volatilization_cation_exchange_factor` is used to determine the volatilization regulator, and has been included in `LayerData` so that it may be modified by the user. This differs from SWAT, which does not allow users to modify it from the default 0.15.
    2) `annual_volatilized_ammonium_total` tracks the total amount of ammonium volatilized from that soil layer in a given year, and is reset by the `do_annual_reset()` method.

Closes #484

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/soil/nitrogen_cycling/nitrification_volatilization.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+"""
+This module handles the nitrification and volatilization operations for the ammonium pool, based on SWAT section 3:1.3.
+"""
+
+
+class NitrificationVolatilization:
+
+    def __init__(self, soil_data: Optional[SoilData] = None, field_size: Optional[float] = None):
+        """This method initializes the SoilData object that this module will work with, or create one if none provided.
```
Comment (ew3361zh) [inline SC_redesign/Crop_and_Soil/soil/nitrogen_cycling/nitrification_volatilization.py]:
```suggestion
        """This method initializes the SoilData object that this module will work with, or creates one if one is not provided.
```

**Review by ew3361zh — APPROVED** [review]
Looks great, Ed. Well documented and great variable/function naming throughout! Test-suite looked good as well. All fairly straight forward code-wise - assume that a SME will review the science side more closely.

I had one grammatical suggestion early on to keep consistent with how things were written in other sections.

File: SC_redesign/Crop_and_Soil/soil/nitrogen_cycling/nitrification_volatilization.py (hunk 1 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/soil/nitrogen_cycling/nitrification_volatilization.py]: Thanks Niko

**Review by varrma — APPROVED** [review]
Great job on this pull request,  Ed ! Well organized and easy to understand.  Clear explanations in your code functions, as Niko mentioned SME opinion on the scientific values or equations will ensure it. Rest all looks good !!

**Pull request merged**